### PR TITLE
feat: Scoped clusters

### PR DESCRIPTION
**Pull request by pasha-codefresh**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: util/argo/argo.go [hunk 1]
```diff
@@ -421,6 +435,10 @@ func GetAppProjectByName(name string, projLister applicationsv1.AppProjectLister
 	for _, repo := range repos {
 		project.Spec.SourceRepos = append(project.Spec.SourceRepos, repo.Repo)
 	}
+	clusters := retrieveScopedClusters(name, db, ctx)
+	for _, cluster := range clusters {
+		project.Spec.Destinations = append(project.Spec.Destinations, argoappv1.ApplicationDestination{Server: cluster.Server, Namespace: "*"})
```
Comment (alexmt) [inline util/argo/argo.go:440]: Cluster might be connected in a "namespaced" mode - `cluster.Namespaces` is not empty.  In this case only these namespaces should be auto-added to the destination. E.g.

```golang
if len(cluster.Namespaces) == 0 {
  project.Spec.Destinations = append(project.Spec.Destinations, argoappv1.ApplicationDestination{Server: cluster.Server, Namespace: "*"})
} else {
  for _, ns := range cluster.Namespaces { 
    project.Spec.Destinations = append(project.Spec.Destinations, argoappv1.ApplicationDestination{Server: cluster.Server, Namespace: ns})
  }
}
```

**Review by alexmt — CHANGES_REQUESTED** [review]

**Review by alexmt — COMMENTED** [review]
Thank you @pasha-codefresh ! Found only one minor issue and it is ready to go

**Review by alexmt — APPROVED** [review]
LGTM

**Comment by sbose78** [comment]
Sorry, is there a PR description or an issue I could read up ? :(

**Comment by alexmt** [comment]
hello @sbose78 . Sure, here it is: https://github.com/argoproj/argo-cd/issues/5283. We also discussed implementation in the proposal: https://github.com/argoproj/argo-cd/blob/master/docs/proposals/project-repos-and-clusters.md

**Pull request merged**